### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by for-ashraf**
In this PR, I have incorporated the following queries/Requirements:

Add a column species of type string to your animals table. Modify your schema.sql file.
Insert the following data:
Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
Remember to add these insertions to your data.sql file.
Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Commit the transaction.
Verify that change was made and persists after commit.
Take a screenshot of the results of your actions.
Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
Inside a transaction:
Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction
Take a screenshot of the results of your actions.
Remember to add all these transactions to your queries.sql file.
Write queries to answer the following questions:
How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?
Take a screenshot of the results of your queries.
Remember to add these queries to your queries.sql file.
Remember to add screenshots with the results of your queries and with the results of your play with transactions to your Pull Request description.
Day2QueryResults

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+insert into animals (id,name,date_of_birth,escape_attempts, neutered, weight_kg) values (4,'Devimon','05-12-2017', 5, true, 11)
+
+/* End Day 1 Queries  */
+
+/* Start Day 2 Queries  */
+
+insert into animals (id,name,date_of_birth,escape_attempts, neutered, weight_kg) values (5,'Charmander','02-08-2020', 0, false, 11)
+insert into animals (id,name,date_of_birth,escape_attempts, neutered, weight_kg) values (6,'Plantmon','11-15-2021', 2, true, 5.7)
+insert into animals (id,name,date_of_birth,escape_attempts, neutered, weight_kg) values (7,'Squirtle','04-02-1993', 3, false, 12.13)
+insert into animals (id,name,date_of_birth,escape_attempts, neutered, weight_kg) values (8,'Angemon','06-12-2005', 1, true, 45)
```
Comment (ShahierNashaat) [inline data.sql]: - [ ] Kindly write the `weight` values right in the insertion of the data.

```sql
INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Charmander', '2020/2/8', '0', 'false', '-11.0');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Plantmon', '2021/11/15', '2', 'true', '-5.7');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Squirtle', '1993/04/02', '3', 'false', '-12.13');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Angemon', '2005/06/12', '1', 'true', '-45.0');

```

File: queries.sql [hunk 2]
```diff
@@ -8,3 +7,36 @@ select name, escape_attempts from animals where weight_kg > 10.5
...
+update animals set weight_kg=weight_kg*-1;
+ROLLBACK to SAVEPOINT delete_animals;
+update animals set weight_kg=weight_kg*-1 where weight_kg < 0;
+COMMIT;
+
+select count(*) from animals;
+select count (*) from animals where escape_attempts = 0
+select avg(weight_kg) from animals;
+select name, escape_attempts,neutered from animals order by escape_attempts desc, neutered;
+SELECT MAX(weight_kg), MIN(weight_kg) FROM animals;
```
Comment (ShahierNashaat) [inline queries.sql]: - [ ] Kindly notice that in this query you should select the minimum and maximum weight of each type of animal.

```sql
SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;
```

File: queries.sql [hunk 3]
```diff
@@ -8,3 +7,36 @@ select name, escape_attempts from animals where weight_kg > 10.5
...
+ROLLBACK to SAVEPOINT delete_animals;
+update animals set weight_kg=weight_kg*-1 where weight_kg < 0;
+COMMIT;
+
+select count(*) from animals;
+select count (*) from animals where escape_attempts = 0
+select avg(weight_kg) from animals;
+select name, escape_attempts,neutered from animals order by escape_attempts desc, neutered;
+SELECT MAX(weight_kg), MIN(weight_kg) FROM animals; 
+SELECT avg(escape_attempts) FROM animals where date_of_birth >= '1990-01-01' AND date_of_birth <  '2000-12-31';
```
Comment (ShahierNashaat) [inline queries.sql]: - [ ] Kindly notice that in this query you should select the average number of escape attempts per animal type of those born between 1990 and 2000.

```sql
SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31';;
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @for-ashraf, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (for-ashraf) [reply queries.sql]: It has been resolved.

File: queries.sql (hunk 2 above)
Comment (for-ashraf) [reply queries.sql]: Yes, you are right. I have incorporated the changes.

File: data.sql (hunk 1 above)
Comment (for-ashraf) [reply data.sql]: I just thought of the negative figures as a dash rather than a negative sign. Now it has been updated, accordingly.

**Review by gemmen29 — APPROVED** [review]
Hi @for-ashraf, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**